frame/uniques/src/lib.rs

```diff
@@ -366,25 +352,14 @@ pub mod pallet {
 			T::ForceOrigin::ensure_origin(origin)?;
 			let owner = T::Lookup::lookup(owner)?;
 
-			ensure!(!Class::<T, I>::contains_key(class), Error::<T, I>::InUse);
-
-			Class::<T, I>::insert(
+			Self::do_create_class(
 				class,
-				ClassDetails {
-					owner: owner.clone(),
-					issuer: owner.clone(),
-					admin: owner.clone(),
-					freezer: owner.clone(),
-					total_deposit: Zero::zero(),
-					free_holding,
-					instances: 0,
-					instance_metadatas: 0,
-					attributes: 0,
-					is_frozen: false,
-				},
-			);
-			Self::deposit_event(Event::ForceCreated(class, owner));
-			Ok(())
+				owner.clone(),
+				owner.clone(),
```

Comment:
clone shouldnt be needed here i think

Reply:
Those clones look fine to me. `do_create_class` takes params by value, the same as the other functions in `functions.rs`.